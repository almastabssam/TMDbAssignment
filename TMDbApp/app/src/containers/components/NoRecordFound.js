import {Text, View} from "react-native";
import React from 'react';
let NoRecordFound =()=>{
    return(
        <View style={{flex: 1, justifyContent: 'center', alignItems: 'center'}}>
            <Text
                style={
                    {
                        height: 40,
                        width: '100%',
                        textAlign: 'center',
                        color: 'gray',
                        fontSize: 18,
                        alignSelf: 'center',
                    }}>
                No Record Found
            </Text>
        </View>
    );
}
export default NoRecordFound;
