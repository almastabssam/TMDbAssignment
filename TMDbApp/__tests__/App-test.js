/**
 * @format
 */
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import APIService from "../app/src/api";
import NoRecordFound from "../app/src/containers/components/NoRecordFound";

describe('Test Render Component Correctly', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<NoRecordFound />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('Test Latest APIs to avoid app crash', () => {
  it('Test API Required Keys', async () => {
    let data = await APIService.getLatestMovies(1);
    expect(true).toEqual(data.results.length?true:false);
    expect(data.results[0]).toHaveProperty('title');
    expect(data.results[0]).toHaveProperty('backdrop_path');
    expect(data.results[0]).toHaveProperty('poster_path');
    expect(data.results[0]).toHaveProperty('overview');
  });
});
describe('Test Movie Detail APIs to avoid app crash', () => {
  it('Test Deatil API Required Keys', async () => {
    let data = await APIService.getMovieDetail(615904);
    expect(data).toHaveProperty('title');
    expect(data).toHaveProperty('backdrop_path');
    expect(data).toHaveProperty('poster_path');
    expect(data).toHaveProperty('overview');
  });
});
