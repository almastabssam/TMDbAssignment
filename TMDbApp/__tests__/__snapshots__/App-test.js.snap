// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Render Component Correctly renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "gray",
        "fontSize": 18,
        "height": 40,
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    No Record Found
  </Text>
</View>
`;
